### ISSUE TITLE
Unqualified field lookup in ON DUPLICATE KEY UPDATE clause doesn't work

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10954]
Given this schema:

```sql
create table t (a int primary key, b int)
```

The following query cannot be translated to e.g. SQL Server:

```sql
insert into t (a, b)
values (1, 2)
on duplicate key update
set b = 2
```

The error on https://www.jooq.org/translate is:

```
Unknown field identifier: [4:7] ...a, b)
values (1, 2)
on duplicate key update
set b [*]= 2
```

This works:

```sql
insert into t (a, b)
values (1, 2)
on duplicate key update
set t.b = 2
```

**Comment by lukaseder** [comment]
In fact, all of these `INSERT` syntaxes are affected, including e.g.:

```sql
insert into t (a, b)
set a = 1, b = 2
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.4 (#10958)
